api/envoy/api/v2/core/protocol.proto

```diff
@@ -34,6 +34,18 @@ message HttpProtocolOptions {
 }
 
 message Http1ProtocolOptions {
+  message HeaderKeyFormat {
+    message ProperCaseWords {
+    }
+
+    oneof header_format {
+      // Formats the header by proper casing words: the first character and any character following
+      // a special character will be capitalized if it's an alpha character. For example,
+      // "content-type" becomes "Content-Type", and "foo$b#$ar" becomes "Foo$B#$Ar".
```

Comment:
@envoyproxy/api-shepherds API LGTM. I think if there are nuances around details with how this is done, e.g. whether it's `-` only or any punctuation, we can sort them out with `ProperCaseWords` fields later.